url9ecf1e8297_nhamidi_ninja_tgz-pJ8-net_fec_openrq_test_TestRunnerJ8/src/Receiver_2.java

```diff
@@ -0,0 +1,349 @@
+
+
+	import java.net.DatagramPacket;
+	import java.net.InetAddress;
+	import java.net.MulticastSocket;
+	import java.net.SocketException;
+	import java.io.IOException;
+	import java.net.UnknownHostException;
+	import java.util.Random;
+
+	public class Receiver_2 {
+		final static int size_int = 4;
+		final static int ACK = -2;
+		final static int NACK = -1;
+		// final static double redondance = 1.00;
+
+		final static int FLAG_PUSH = 999999999;
+		final static int FLAG_STOP = 111111111;
+
+		// Simulate a lost of the canal
+		public static boolean packet_is_loss(float percentageLoss) {
+			Random randomGenerator = new Random();
+			int rand = randomGenerator.nextInt(100);
+			if (rand < (int) percentageLoss) {
+				return true;
+			}
+			return false;
+
+		}
+
+		public static final int byte_array_to_int(byte[] byte_array) {
+			int interger=0;
+			for (int i = 0; i < size_int; i++) {
+				interger = (interger << 8) + (byte_array[i] & 0xff);
+			}
+			return interger;
+		}
+
+		public static byte[] parameter_test(byte[] receivePacket, int parameter_test) {
+
+			byte[] packetData = receivePacket;
+
+			switch (parameter_test) {
+			case 1:
+
+				break;
+			case 10:
+
+				break;
+			case 20:
+
+				break;
+			default:
+			}
+
+			return packetData;
+		}
+
+		public static final int look_for_the_answer(InetAddress groupeIP, int port)
+				throws IOException {
+			int packetData = 0;
+			MulticastSocket socketReception;
+			socketReception = new MulticastSocket(port);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *socketReception*. The resource is referenced at lines 64, 70. The resource is closed at line 86. There are other execution paths that do not contain closure statements, for example, when MulticastSocket.joinGroup throws an exception. Ensure *socketReception* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).